executor/modes/local/handlers.py

```
@@ -132,3 +134,244 @@ async def handle_connect_error(self, data: Any) -> None:
+        finally:
+            with self._upgrade_lock:
+                self._upgrade_in_progress = False
+
+    def _execute_upgrade_sync(
+        self,
+        force: bool,
+        auto_confirm: bool,
+        verbose: bool,
+        registry: Optional[str],
+        registry_token: Optional[str],
+    ) -> "UpdateResult":
```

Ruff (0.15.6)

[error] Undefined name `UpdateResult`

(F821)